Description: fix dtl bad const
Author: mirabilos <tg@debian.org>
Forwarded: https://github.com/cubicdaiya/dtl/pull/18

--- a/thirdparty/dtl/dtl/Diff.hpp
+++ b/thirdparty/dtl/dtl/Diff.hpp
@@ -164,7 +164,7 @@ namespace dtl {
             return trivial;
         }
         
-        void enableTrivial () const {
+        void enableTrivial () {
             this->trivial = true;
         }
         
