Origin: upstream, commit:24e239ed74770e24d2981883d5d435302735567a
Author: Dmitri Ovodok <dmitrio95@yandex.ru>
Description: Fix a crash on removing HBox from a score

--- a/libmscore/box.cpp
+++ b/libmscore/box.cpp
@@ -457,9 +457,12 @@ void HBox::layout()
             setPos(x, y);
             bbox().setRect(0.0, 0.0, w, h);
             }
-      else {
+      else if (system()) {
             bbox().setRect(0.0, 0.0, point(boxWidth()), system()->height());
             }
+      else {
+            bbox().setRect(0.0, 0.0, 50, 50);
+            }
       Box::layout();
       }
 
