Origin: upstream, commit:5301009829215619953d33f03932b09c3ead5e7b
Author: Joachim Schmitz <jojo@schmitz-digital.de>
Description: fix #300738: Ottava (8va/8vb) number is too big

--- a/libmscore/style.cpp
+++ b/libmscore/style.cpp
@@ -447,7 +447,7 @@ static const StyleType styleTypes[] {
       { Sid::ottavaLineStyle,         "ottavaLineStyle",         QVariant(int(Qt::DashLine)) },
       { Sid::ottavaNumbersOnly,       "ottavaNumbersOnly",       true },
       { Sid::ottavaFontFace,          "ottavaFontFace",          "FreeSerif" },
-      { Sid::ottavaFontSize,          "ottavaFontSize",          12.0 },
+      { Sid::ottavaFontSize,          "ottavaFontSize",          10.0 },
       { Sid::ottavaFontSpatiumDependent, "ottavaFontSpatiumDependent", true },
       { Sid::ottavaFontStyle,         "ottavaFontStyle",         int(FontStyle::Normal) },
       { Sid::ottavaColor,             "ottavaColor",             QColor(0, 0, 0, 255) },
